core/src/main/java/jenkins/mvn/SettingsProvider.java

```diff
@@ -46,17 +96,29 @@ public static SettingsProvider parseSettingsProvider(StaplerRequest req) throws
      * @param listener
      *            the listener of the current build
      * @return the path to the settings.xml
+     * @deprecated directly invoke {@link #supplySettings(Run, FilePath, TaskListener)}
      */
+    @Deprecated
     public static final FilePath getSettingsFilePath(SettingsProvider settings, AbstractBuild<?, ?> build, TaskListener listener) {
-        FilePath settingsPath = null;
-        if (settings != null) {
-            settingsPath = settings.supplySettings(build, listener);
+        FilePath settingsPath;
+        if (settings == null) {
+            settingsPath = null;
+        } else {
+            try {
+                settingsPath = settings.supplySettings(build, build.getWorkspace(), listener);
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to prepare Maven settings.xml for " + build +
```

Comment:
🐛 No RunTime exceptions please, it may cause random fallout. Log & return `null`

Reply:
What do you mean by random fallout. Wont this just fail the build?